ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/AwsConnectorExecutor.java

```diff
@@ -68,7 +71,27 @@ public AwsConnectorExecutor(Connector connector, SdkHttpClient httpClient) {
     }
 
     public AwsConnectorExecutor(Connector connector) {
-        this(connector, new DefaultSdkHttpClientBuilder().build());
+        super.validate(connector);
+        this.connector = (AwsConnector) connector;
+        Duration connectionTimeout = Duration.ofMillis(super.getHttpClientConfig().getConnectionTimeout());
+        Duration readTimeout = Duration.ofMillis(super.getHttpClientConfig().getReadTimeout());
+        try (
```

Comment:
aren't adding a catch for this try? Also there is the existing constructor "public AwsConnectorExecutor(Connector connector, SdkHttpClient httpClient)". Should we apply default client configs for this httpClient too for the sake of BWC?

Reply:
added exception.